xunit-autolabeler-v2/ast_parser/lib/file_utils.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2020 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import List, Callable
+
+import os
+import subprocess
+
+import constants
+
+
+def _getFiles(root_dir: str, predicate: Callable[[str], bool]) -> List[str]:
+    """Recursively list the files in a given directory
+       whose names match the provided predicate function
+
+    Args:
+        root_dir: the root directory to search from
+        predicate: the predicate function
+                                       to filter filenames with
+
+    Returns:
+        A list of filepaths relative to root_dir that match the predicate
+    """
+    if os.path.basename(os.path.normpath(root_dir)).startswith('.'):
+        # Ignore dot-directories
+        return []
+
+    paths = [os.path.join(root_dir, path) for path in os.listdir(root_dir)
+             if path != 'lib']
```

Comment:
I still don't think it's right, even for python.

I know we should exclude it for appengine samples, but what about other samples?

Also, what about other languages?

You're passing in `predicate` function. At least can we exclude it with `predicate`?

Reply:
Done.